agent/command/generate.go

```diff
@@ -103,14 +103,21 @@ func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, lo
 		pollRetryMaxDelay = 15 * time.Second
 	)
 
-	var generateStatus *apimodels.GeneratePollResponse
 	err = utility.Retry(
 		ctx,
 		func() (bool, error) {
-			generateStatus, err = comm.GenerateTasksPoll(ctx, td)
+			generateStatus, err := comm.GenerateTasksPoll(ctx, td)
 			if err != nil {
 				return true, err
 			}
+			// only retry if the errors are with saving the project
+			if len(generateStatus.Errors) > 0 {
+				fullErr := errors.New(strings.Join(generateStatus.Errors, ", "))
+				if strings.Contains(fullErr.Error(), evergreen.RetryGenerateTasksError) {
+					return true, fullErr
```

Comment:
Could you explain why this is the error case in which we _should_ retry?

Reply:
I'm specifically wrapping errors from Save with this error message. Project validation errors will typically come from NewVersion or CheckProjectConfigurationIsValid, and from poking around splunk i think the only real errors come from Save (with system errors or etc).

Reply:
Doesn't this mean that if we ever error from Save, we'll retry? https://github.com/evergreen-ci/evergreen/pull/4758/files#diff-fbf882d1f7202d7e200578a497a157e6f7273b8d2195ffbc7d7882fe88472069L187

I'm curious why that's the right behavior.

Reply:
Yep we will. The only errors I think we should suppress come from building/validating the project, which happens before Save. If you look at some of the errors I linked in the splunk logs, some errors we get from save are:
```
errors adding new tasks: connection(evergreendb-3.10gen-mci.4085.mongodbdns.com:27017[-203]) socket was unexpectedly closed: EOF
error upserting parser project '60a2957e9ccd4e597bc5b2bc': database error upserting parser project '60a2957e9ccd4e597bc5b2bc': (NotMaster) not master
```

which come from both of the primary functions in Save, and seem worth retrying. Note that we still don't wrap results not found or duplicate key, since those are a special case.

Reply:
Oh, that makes sense. Thanks!